src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health = 100;
+  constructor(name) {
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name) {
+    super(name);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name) {
+    super(name);
+  }
+  bite(obj) {
+    if (obj.hidden === false) {
+      obj.health -= 50;
+    }
+
+    Animal.alive.forEach((elem) => {
+      if (elem.health > 0) {
+        return elem;
+      }
+      Animal.alive.splice(Animal.alive.indexOf(elem), 1);
```

Comment:
There is a potential issue with modifying the `Animal.alive` array while iterating over it. This can lead to unexpected behavior or errors. Consider using a different approach, such as filtering the array after the loop, to remove dead animals.